addons/pos_sale/models/sale_order.py

```diff
@@ -26,6 +27,20 @@ def action_view_pos_order(self):
             'domain': [('id', 'in', linked_orders.ids)],
         }
 
+    @api.depends('invoice_ids.amount_total_signed', 'invoice_ids.state', 'pos_order_line_ids.price_subtotal_incl')
+    def _compute_amount_unpaid(self):
+        for sale_order in self:
```

Comment:
here you should just do and it should do the trick and depends on the lines:

sale_order.amount_unpaid = sale_order.amount_total - sum(sale_order.order_line.filtered(lambda l: l.is_downpayment).price_total)